### PR TITLE
Update ember packages and 3rd party depenencies

### PR DESCRIPTION
**Pull request by jaredgalanis**
This bumps all Ember packages to the latest LTS (Ember 4.8). This also upgrades any other packages necessary to get there.

There was quite a lot to change to get things building and tests passing again, but much of it was syntax.

This also removes a lot of packages and cruft that weren't being used, but it does not update every single package as that was not necessary to get to Ember 4.

The following is a list of packages we are not updating in this PR:

We're not updating these because they did not HAVE to be updated to get to Ember 4 and it is unclear what steps are needed to get to the latest. This should be ticketed in future work.
 ◯ ajv                  latest  6.12.0  ❯  8.12.0  https://ajv.js.org
 ◯ sweetalert2          latest  7.33.1  ❯  11.7.3  https://sweetalert2.github.io/

We're not updating bootstrap because I want to avoid slient style changes and because we might want to focus energy in moving to tailwind-css and away from bootstrap entirely.
 ◯ bootstrap            latest  4.6.2   ❯  5.2.3   https://getbootstrap.com/

We're not updating ember core packages past 4.8 because that is the latest LTS.
 ◯ ember-cli            latest  4.8.0   ❯  4.11.0  https://cli.emberjs.com/release/
 ◯ ember-data           latest  4.7.3   ❯  4.11.3  https://github.com/emberjs/data#readme
 ◯ ember-source         latest  4.8.4   ❯  4.11.0  https://emberjs.com/

We're not updating ember-template-lint because there are a host of rules that need to be ignored and then resolved. This can be done in subsequent tickets.
 ◯ ember-template-lint  latest  4.18.2  ❯  5.7.1   https://github.com/ember-template-lint/ember-template-lint

There is a separate branch to remove ember-concurrency-decorators. 
 ◯ ember-concurrency-decorators  latest  1.0.0   ❯  2.0.3   https://github.com/machty/ember-concurrency-decorators#readme

### CODE REVIEW

File: .ember-cli.js [hunk 1]
```diff
@@ -13,6 +9,11 @@ module.exports = {
   */
   liveReload: false,
   yarn: true,
-  watcher: process.env.WATCHER || 'polling',
+  // watcher: process.env.WATCHER || 'polling',
```
Comment (jaredgalanis) [inline .ember-cli.js:12]: do we still need this?

File: app/components/nav-bar/index.hbs [hunk 2]
```diff
@@ -71,15 +71,15 @@
             >
               <span class="pr-1">
                 {{! template-lint-disable require-presentational-children }}
-                <GravatarImage
```
Comment (jaredgalanis) [inline app/components/nav-bar/index.hbs:74]: need to address this

File: app/components/workflow-basics/index.hbs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {{! template-lint-disable no-action no-triple-curlies no-unknown-arguments-for-builtin-components require-button-type require-input-label simple-unless }}
 {{yield}}
-<div {{did-insert this.setupDoiJournal}} {{did-insert this.setupSubmission}} {{did-update this.setupSubmission}}></div>
+{{!-- <div {{did-insert this.setupDoiJournal}} {{did-insert this.setupSubmission}} {{did-update this.setupSubmission}}></div> --}}
 {{#unless (or @submission.id @preLoadedGrant)}}
```
Comment (jaredgalanis) [inline app/components/workflow-basics/index.hbs:4]: investigate whether we fixed this

File: app/components/workflow-basics/index.js [hunk 4]
```diff
@@ -66,8 +69,8 @@ export default class WorkflowBasics extends Component {
   @action
   setupSubmission() {
     if (!this.isProxySubmission) {
-      set(this, 'submission.submitter', get(this, 'currentUser.user'));
-      set(this, 'submission.preparers', A());
+      this.submission.submitter = this.currentUser.user;
+      // this.submission.preparers = [];
```
Comment (jaredgalanis) [inline app/components/workflow-basics/index.js:73]: make sure this is ok

File: app/components/workflow-grants/index.js [hunk 5]
```diff
@@ -85,6 +84,27 @@ export default class WorkflowGrants extends Component {
   constructor() {
     super(...arguments);
 
+    // if (this.args.preLoadedGrant) {
```
Comment (jaredgalanis) [inline app/components/workflow-grants/index.js:87]: delete this

File: app/templates/application.hbs [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {{! template-lint-disable no-inline-styles require-valid-alt-text }}
-<EmberLoadRemover />
+{{! <EmberLoadRemover /> }}
```
Comment (jaredgalanis) [inline app/templates/application.hbs:2]: Does this need fixing?

**Review by jabrah — APPROVED** [review]
We talked through the PR on a call. Looking good now. Will be good to run the pass-acceptance-tests in pass-docker with this updated code

**Pull request merged**